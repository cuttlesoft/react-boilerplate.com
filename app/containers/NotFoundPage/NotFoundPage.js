/**
 * NotFoundPage
 *
 * This is the page we show when the user visits a url that doesn't have a route
 */

import React from 'react'
import { FormattedMessage } from 'react-intl'

import { Header } from 'components/Header'
import messages from './NotFoundPage.messages'

export default function NotFound() {
  return (
    <article>
      <Header level="1">
        <FormattedMessage {...messages.header} />
      </Header>
    </article>
  )
}
