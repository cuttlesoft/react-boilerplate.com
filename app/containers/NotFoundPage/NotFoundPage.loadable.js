import React from 'react'

// Components
import { LoadingIndicator } from 'components/LoadingIndicator'

// Helpers
import loadable from 'utils/loadable'

/**
 * Asynchronously loads the component for NotFoundPage
 */
export default loadable(() => import('./NotFoundPage'), {
  fallback: <LoadingIndicator />,
})
