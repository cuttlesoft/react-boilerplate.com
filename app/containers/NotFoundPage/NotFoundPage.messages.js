import { defineMessages } from 'react-intl'

export const scope = 'boilerplate.containers.NotFoundPage'

/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Page not found.',
  },
})
