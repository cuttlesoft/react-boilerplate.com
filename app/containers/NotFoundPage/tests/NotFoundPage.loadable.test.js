/**
 * Test the HomePage
 */
import React from 'react'
import { waitForElement, render } from '@testing-library/react'
import NotFoundPage from '../NotFoundPage.loadable'

test('should lazy load and match the snapshot', async () => {
  const { container } = render(<NotFoundPage />)
  const lazyComponent = await waitForElement(() => container)
  await expect(lazyComponent).toMatchSnapshot()
})
