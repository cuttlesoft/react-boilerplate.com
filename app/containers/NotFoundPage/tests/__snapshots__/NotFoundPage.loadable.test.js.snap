// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should lazy load and match the snapshot 1`] = `
<div>
  <article>
    <h1
      class="StyledHeading-sc-1rdh4aw-0 rfVJR"
    >
      <span>
        Page not found.
      </span>
    </h1>
  </article>
</div>
`;
